codebase-pipelines/artifactstore.tf

```diff
@@ -0,0 +1,92 @@
+resource "aws_s3_bucket" "artifact_store" {
+  # checkov:skip=CKV_AWS_144: Cross Region Replication not Required
+  # checkov:skip=CKV2_AWS_62: Requires wider discussion around log/event ingestion before implementing. To be picked up on conclusion of DBTP-974
+  # checkov:skip=CKV2_AWS_61: This bucket is only used for the pipeline artifacts, so no requirement for lifecycle configuration
+  # checkov:skip=CKV_AWS_21: This bucket is only used for the pipeline artifacts, so no requirement for versioning
+  # checkov:skip=CKV_AWS_18:  Requires wider discussion around log/event ingestion before implementing. To be picked up on conclusion of DBTP-974
+  bucket = "${var.application}-${var.codebase}-codebase-pipeline-artifact-store"
+
+  tags = local.tags
+}
+
+data "aws_iam_policy_document" "artifact_store_bucket_policy" {
+  statement {
+    principals {
+      type        = "*"
+      identifiers = ["*"]
```

Comment:
Could these be 
```suggestion
      type        = "Service"
      identifiers = ["codepipeline.amazonaws.com"]
```

Reply:
This is a deny all policy to ensure SSL only access